doc/Programming-guide/Key-platform-features/Devices-provisioning-and-registration/index.md

```diff
@@ -9,77 +9,75 @@ sort_idx: 10
+This section discusses the process of endpoint authentication and the functionality of the Kaa Credentials service.
 
-Kaa uses hybrid encryption system that is based on RSA with 2048-bit key pair and AES with 256(512)-bit key. 
-During the start of new session, endpoint delivers encrypted session key and the digital signature of this key.
-Session key is a randomly generated AES key that is also encrypted with public key of the kaa node which serves this session or request. 
-Session key signature is based on the private key of the endpoint.
-In order to authenticate endpoint and validate integrity of the request, kaa node validates signature of the session key.
+## Endpoint authentication
 
-Although other authentication strategies are possible they are out of scope of this article.
+Kaa uses hybrid encryption system that is based on RSA with 2048-bit key pair and AES with 256(512)-bit key.
+During the start of a new session, every endpoint sends an encrypted and digitally signed session key.
+Session key is a randomly generated AES key that is also encrypted with public key of the Kaa node that handles this session or request.
+The digital signature of a session key is based on the private key of the endpoint.
+To authenticate an endpoint and validate its request, Kaa node validates the digital signature of the session key.
 
-## Endpoint Registration
+Although other authentication strategies are possible, they are out of scope of this section.
 
-The endpoint registration is the process of initial communication setup that ensures that the endpoint is registered within the Kaa cluster 
-along with the corresponding security credentials and the endpoint profile. 
-It is not until the endpoint registers on the server that the Kaa framework services become available to the client application. 
-During the process of registration, the Kaa endpoint at first communicates with one or multiple Bootstrap services to obtain a list of available Operations services, 
-and then with the Operations service to submit the endpoint data and complete the registration.
+## Endpoint registration
 
+To function within a [Kaa cluster]({{root_url}}Glossary/#kaa-cluster), every endpoint must be registered using the security credentials and the endpoint profile.
+During the registration process, the endpoint first communicates with one or multiple [Bootstrap services]({{root_url}}Glossary/#bootstrap-service) to obtain a list of available [Operations services]({{root_url}}Glossary/#operations-service).
+Then it communicates with the Operations service to submit the endpoint data and complete the registration.
+
+The endpoint registration process is illustrated in the picture below.
 
 ![Endpoint Registration](attach/registration.png)
 
-### Operations service list resolution
+### Operations services list
```

Comment:
It is grammatically correct but to make it sound better, I'll change it to "List of Operations services".